boards.txt

```diff
@@ -1051,6 +1087,10 @@ GenF4.menu.upload_method.hidMethod.upload.tool=hid_upload
 GenF4.menu.upload_method.hidMethod.build.flash_offset=0x4000
 GenF4.menu.upload_method.hidMethod.build.bootloader_flags=-DBL_HID -DVECT_TAB_OFFSET={build.flash_offset}
 
+GenF4.menu.upload_method.MassStorage=Mass Storage
```

Comment:
This is not a valid upload method for Generic menu.

Reply:
Because no other programmers than the on-board STlink on the Nucleo boards have this option? I've been using this option while I've worked on the new pinout.

Reply:
Yes. I know you use the Nucleo to validate but this is not a valid options for all other boards.
Keep in mind this is a _Generic_ Core and It have to be as generic as possible. If I add this option and user select the Black F407, this will not be functional... then I will have to explain why it is not... again and again.
If an user wants some customization for his convenience then he can:
 * https://github.com/stm32duino/wiki/wiki/Custom-board-based-on-a-core
 * use platform.local.txt
 * ...